google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/CloudSdkServiceManager.java

```diff
@@ -139,8 +139,9 @@ private void doWait(
       String progressMessage,
       CloudSdkStatusHandler sdkLogging) {
     CloudSdkService cloudSdkService = CloudSdkService.getInstance();
-    // if not ready, attempt to fix and install now.
-    if (!isSdkReady() && cloudSdkService.isInstallSupported()) {
+    // if install supported, make sure SDK is in good shape and installed.
+    // even if status is READY, this will double-check the current state from the last install.
+    if (cloudSdkService.isInstallSupported()) {
```

Comment:
so it can be READY, but in a bad state?

Reply:
correct, please see one possible scenario in #1993.

Reply:
I see. so install() itself will then check its status and install if it corrupted/removed/etc, correct?

Also since you are lumping a few fixes together here, is this line the only fix in support of #1993?

Reply:
correct, this should make it double check its status before the actual operation, so it was simpler than I originally thought.

Reply:
Yes, your understanding is totally correct.
As for configuration checking, we do not mark deployment/run configuration as invalid when SDK supports self-install at all, since we plan to install and fix it at the point of deployment (if it's broken). So for managed SDK it doesn't buy us anything to re-check the status at those points.

Reply:
oh, so if its installed, but broken, the then the configuration will be marked as OK; but then upon deploy/run, it will be re-installed if broken?

I think that makes sense, especially since the configuration error checks should be lightweight.

Reply:
Exactly, you're right.